Documentation/BasicOperators.md

```diff
@@ -46,41 +46,394 @@ code-style will be used.
 ## Performing side effects with signals
 
 ### Observation
+
+Signals can be observed with the `observe` function. It takes a `Sink` as argument to which any future events are sent. 
+
+```Swift
+signal.observe(Signal.Observer { event in
+    switch event {
+    case let .Next(next):
+        println("Next: \(next)")
+    case let .Error(error):
+        println("Error: \(error)")
+    case .Completed:
+        println("Completed")
+    case .Interrupted:
+        println("Interrupted")
+    }
+})
+```
+
+Alternatively, callbacks for the `Next`, `Error`, `Completed` and `Interrupted` events can be provided which will be called when a corresponding event occurs.
```

Comment:
Can you append another sentence that each param is optional and only those events you care about need callbacks.